src/main/java/org/springframework/data/repository/query/Parameters.java

```diff
@@ -62,11 +62,11 @@ public Parameters(Method method) {
 
 		Assert.notNull(method);
 
-		this.parameters = new ArrayList<T>();
-		this.dynamicProjectionIndex = -1;
-
 		List<Class<?>> types = Arrays.asList(method.getParameterTypes());
 
+		this.parameters = new ArrayList<T>(types.size());
```

Comment:
I need the types variable here in order to retrieve the size. If I call method.getParameters() again I have the overhead of cloning the method parameters, which I wanted to avoid. As far as I saw, you're not yet compiling against Java 8, which would have enabled me to call Method.getParameterCount().